hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/HandleWorkflow.java

```diff
@@ -678,12 +679,15 @@ private PreHandleResult addMissingSignatures(
 
         // prepare signature verification
         final var verifications = new HashMap<Key, SignatureVerificationFuture>();
-        final var payerKey = previousResult.payerKey();
-        verifications.put(payerKey, previousResult.verificationResults().get(payerKey));
+        final var payer = solvencyPreCheck.getPayerAccount(storeFactory, previousResult.payer());
+        final var payerKey = payer.key();
 
         // expand all keys
         final var expanded = new HashSet<ExpandedSignaturePair>();
         signatureExpander.expand(sigPairs, expanded);
+        if (payerKey != null && !HapiUtils.isHollow(payer)) {
+            signatureExpander.expand(payerKey, sigPairs, expanded);
+        }
```

Comment:
I plan to do a follow-up PR that makes sure we handle hollow accounts and invalid transactions without a payer key correctly. In this upcoming PR, I also want to remove the `payerKey` from `PreHandleResult`.

This fix is about expanding the payer key instead of using it directly, as this works only for primitive keys.